### PR TITLE
generate-ci-config should use proper equality check for strings

### PR DESCRIPTION
**Pull request by lberk**

### CODE REVIEW

**Comment by lberk** [comment]
@matzew sorry? I'm not sure what you mean?

File: openshift/ci-operator/generate-ci-config.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ branch=${1-'knative-v0.6.0'}
 openshift=${2-'4.3'}
 promotion_disabled=${3-false}
 
-if [[ "$branch" -eq "knative-next" ]]; then
+if [[ "$branch" == "knative-next" ]]; then
```
Comment (matzew) [inline openshift/ci-operator/generate-ci-config.sh:7]: should the branch name be `release-next`, instead of `knative-next` ?

Comment (matzew) [reply openshift/ci-operator/generate-ci-config.sh:7]: @lberk   ^

Comment (lberk) [reply openshift/ci-operator/generate-ci-config.sh:7]: @matzew 
no, based on the way the branch/version variable is derived and passed in `update-ci` this is correct

**Comment by matzew** [comment]
why do we pass in `knative-next` ?  the "CI / nightly" branch is called `release-next`, isnt that a bit confusing? :thinking:

**Comment by lberk** [comment]
@matzew sure, but isn't that a different issue? this PR just allows the existing check to actually work.

**Pull request merged**